boa/src/syntax/lexer/mod.rs

```diff
@@ -81,14 +99,16 @@ where
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub(crate) enum InputElement {
     Div,
-    RegExp,
-    RegExpOrTemplateTail,
-    TemplateTail,
+    _RegExp,
+    _RegExpOrTemplateTail,
+    _TemplateTail,
 }
 
 impl Default for InputElement {
     fn default() -> Self {
-        todo!("what is the default input element?")
+        InputElement::Div
+        // Decided on InputElementDiv as default for now based on documentation from
+        // <https://tc39.es/ecma262/#sec-ecmascript-language-lexical-grammar>
```

Comment:
If I understand the spec correctly, the default one cannot be `Div`, since by default, if you find `/` as the first parameter for the lexer, this means the beginning of a RegExp literal:
```javascript
/hello/.test("hello");
```

Reply:
Hmmm, to be honest, I'm not really sure what should be default - I'll look more into this.

Reply:
I have changed this to InputElementRegExpOrTemplateTail based on the spec at https://tc39.es/ecma262/#sec-ecmascript-language-lexical-grammar as using Div has the problems as you say and InputElementRegExp can't be used as it would allow the code to contain a singular RightBracePunctuator which testing on js-fiddle it doesn't.